src/main/java/kleebot/storage/Storage.java

```diff
@@ -0,0 +1,92 @@
+package kleebot.storage;
+
+import kleebot.task.Deadline;
+import kleebot.task.Event;
+import kleebot.task.Task;
+import kleebot.task.ToDo;
+
+import java.io.*;
+import java.util.*;
+
+public class Storage {
+
+    private String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    public ArrayList<Task> load() throws FileNotFoundException {
+        ArrayList<Task> tasks = new ArrayList<>();
+//        File dir = new File("./KleeData");
+        File file = new File(filePath);
+        // if (dir.exists()) // not sure if this will still be needed?
+        if (file.exists()) {
+            try {
+                Scanner reader = new Scanner(file);
+                while (reader.hasNextLine()) {
+                    String data = reader.nextLine();
+                    if (data.isEmpty()) break;
+                    String[] splitted = data.split("\\s+\\|\\s+"); // regex splits by " | " hopefully
+                    String type = splitted[0];
+                    boolean done = splitted[1].equals("1");
+
+                    switch (type) {
+                        case "T":
+                            ToDo tmp = new ToDo(splitted[2]);
+                            if (done) { tmp.markAsDone(); }
+                            tasks.add(tmp);
+                            break;
```

Comment:
Perhaps use more descriptive names for the ToDo object and boolean `done`?
Same thing noticed for the other Task type objects below.
```suggestion
                            ToDo todo = new ToDo(splitted[2]);
                            if (isDone) { 
                                tmp.markAsDone();
                            }
                            tasks.add(todo);
                            break;
```